OpenRA.Game/Primitives/ActionQueue.cs

```diff
@@ -17,37 +18,56 @@ namespace OpenRA.Primitives
 	/// </summary>
 	public class ActionQueue
 	{
-		object syncRoot = new object();
-		PriorityQueue<DelayedAction> actions = new PriorityQueue<DelayedAction>();
+		readonly List<DelayedAction> actions = new List<DelayedAction>();
 
-		public void Add(Action a) { Add(a, 0); }
-		public void Add(Action a, int delay)
+		public void Add(Action a, int desiredTime)
 		{
-			lock (syncRoot)
-				actions.Add(new DelayedAction(a, Game.RunTime + delay));
+			if (a == null)
+				throw new ArgumentNullException("a");
```

Comment:
The first string is the name of the argument. This will automatically translate into a suitable error message, because of the exception type. http://www.dotnetperls.com/argumentexception